js/misc/paygManager.js

```diff
@@ -88,6 +112,57 @@ function _isPaygEnabled() {
     return false;
 }
 
+// Takes an UNIX timestamp (in seconds) and returns a string
+// with a precision level appropriate to show to the user.
+//
+// The returned string will be formatted just in seconds for times
```

Comment:
I'm okay with this decision for now, though I still would consider simplifying to just always show "XX days YY hours ZZ minutes".